syft/execution/plan.py

```diff
@@ -682,12 +777,17 @@ def tag_sort(keyword):
     """
     Utility function to sort tensors by their (unique) tag including "keyword"
     """
-    # TODO is only works up to 9 return values, because comparison is done on str and '7' > '16'
+
     def extract_key(placeholder):
         for tag in placeholder.tags:
             if keyword in tag:
-                return tag
-
-        return TypeError(f"Tag '{keyword}' not found in placeholder tags:", placeholder.tags)
+                try:
+                    return int(tag.split("-")[-1])
```

Comment:
I see a lot of logic about tags appearing, would it be worth having a Tag class to avoid spreading it everywhere?

Reply:
I agree, but I'm not sure tags are mature enough for this: here I use tags with a common prefix and number index appended. I don't know if all tags will support this features or not, so keeping this hidden in plan.py doesn't bother me.